src/components/FormInput/FormInput.styles.ts

```diff
@@ -0,0 +1,69 @@
+import { TextField } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+import { SIZE_CONFIGS } from './FormInput.constants';
+
+import type { StyledFormInputProps } from './FormInput.types';
+
+export const StyledFormInput = styled(TextField, {
+  shouldForwardProp: (prop) => (typeof prop === 'string' ? !prop.startsWith('$') : true),
+})<StyledFormInputProps>(({ theme, $variant, $size, $error }) => {
+  const sizeConfig = $size ? SIZE_CONFIGS[$size] : SIZE_CONFIGS['medium'];
+
+  return {
+    width: '100%',
+    '& .MuiInputBase-root': {
+      height: theme.spacing(Number(sizeConfig.height.replace('rem', '')) * 2),
```

Comment:
The height is computed by parsing a rem string and feeding an ad-hoc Number(...) * 2 into theme.spacing: height: theme.spacing(Number(sizeConfig.height.replace('rem', '')) * 2). This is brittle (string manipulation), incorrect semantics (theme.spacing expects a spacing factor, not a rem), and will break if SIZE_CONFIGS uses different units.

**Sugerencia:**
```typescript
Use numeric tokens in SIZE_CONFIGS (see suggestions in constants file) and compute height consistently. Example using numeric rem tokens and converting to px/rem properly:

const heightRem = sizeConfig.heightRem; // number
height: `${heightRem}rem`,

Or derive using theme.spacing if you store spacing units: height: theme.spacing(heightUnits).

Avoid parsing strings in styles.
```